src/input_formatter/expression_formatter.ts

```diff
@@ -0,0 +1,147 @@
+import { addIssue } from "../issueHandling/issueMapping";
+const allowedExpressionToken = /(bracketLeft|bracketRight|equivalence|implicationRight|implicationLeft|negation|conjunction|disjunction|equal|addition|subtraction|division|multiplication)+/;
+const logicConnector = /(conjunction|disjunction|equivalence|implicationRight|implicationLeft)+/;
+const nonLogicToken = /(bracketLeft|bracketRight|negation)+/;
+
+/**
+	* ---------CurrentExpressionFormatter------------
+	* @param expression written by a user
+	* @return a formatted expression as String
+	*/
+export function expressionFormatter(expression: string) {
+	const preFormattedExpression: string[] = preFormatExpressionFromImput(expression);
+	errorDetector(preFormattedExpression);
+	// TO DO SOME STUFF WITH THE TOKENS AND THE ARRAY; FORMAT THE ARRAY AND CATCH ERRORS
+	const finalFormattedExpression: string[] =
+		replaceExpressionElementsIntoPrologCode(preFormattedExpression);
+	return finalFormattedExpression;
+}
+
+/**
+ * 
+ * @param expression as a input string typed by user
+ * @return a preformatted expression where some different input styles for logical vocabulary gets formatted into one single style
+ */
+export function preFormatExpressionFromImput(expression: string) {
+	const formattedInputExpression: string = expression
+		.replace(/(\$)/g, "")
+		.replace(/(\[|\()/g, " bracketLeft ")
+		.replace(/(\]|\))/g, " bracketRight ")
+		.replace(/(<-->|<==>|<=>|<->)/g, " equivalence ")
+		.replace(/(->|-->|=>|==>|\\rightarrow)/g, " implicationRight ")
+		.replace(/(<-|<--|<=|<==|\/leftarrow)/g, " implicationLeft ")
+		.replace(/(neq|not|nicht|¬|-|\\neg)/ig, " negation ")
+		.replace(/(and|und|&|∧|\/\\)/ig, " conjunction ")
+		.replace(/(or|oder|\||∨|\\\/)/ig, " disjunction ")
+		.replace(/(=|==|===|gleich)/g, " equal ")
+		.replace(/(\+|plus|add|sum)/g, " addition ")
+		.replace(/(-|minus)/g, " subtraction ")
+		.replace(/(:|\/|div|geteilt)/g, " division ")
+		.replace(/(\*|mal|mult)/g, " multiplication ")
+		.trim()
+		.replace(/\s{1,}/g, ",");
+	const splittedExpression: string[] = formattedInputExpression.split(/[,]+/g);
+	return splittedExpression;
+
+}
+
+/**
+	* @param formattedExpression
+	* @return finalExpression where exrpession elements got replaced with readable prolog code elements
+	*/
```

Comment:
*'s are not alligned. But that may have to do with browserdisplaystyle of github.